### PR TITLE
Refactoring the Spec in-image examples 

### PR DESCRIPTION
**Pull request by jordanmontt**
Done things

- Created dynamic layout example
- Moved all defaultLayout methods from class side to instance side
- Recategorised the classes inside the package Spec2-Examples
- Reordered the examples
- Added filtering list example
- Moved the form example to the right package
- Refactored the transmission example and added it to the spec demo presenter

Enregistrement-de-lecran-2022-03

### CODE REVIEW

**Comment by Ducasse** [comment]
Oh this is great!!!

File: src/Spec2-Examples/SpApplicationWithToolbar.class.st [hunk 1]
```diff
@@ -42,6 +34,15 @@ SpApplicationWithToolbar >> addItemTo: aGroup [
 	self build
 ]
 
+{ #category : #layout }
+SpApplicationWithToolbar >> defaultLayout [
+
+	^ SpBoxLayout newTopToBottom
+		add: menu withConstraints: [ :constraints | constraints height: self class toolbarHeight ];
```
Comment (estebanlm) [inline src/Spec2-Examples/SpApplicationWithToolbar.class.st]: in fact... add:height: would be not correct here (`add: ... expand: false` would be)

**Review by estebanlm — CHANGES_REQUESTED** [review]
thanks for this, is very needed!
can you replace the bad usages of add:withConstraints: to fix a height with add:expand: false, as a last step?

File: src/Spec2-Examples/SpApplicationWithToolbar.class.st (hunk 1 above)
Comment (jordanmontt) [reply src/Spec2-Examples/SpApplicationWithToolbar.class.st]: Ok, I'll do that

**Comment by jordanmontt** [comment]
I push changes in my branch but the PR is not updating... Do you have any idea? @estebanlm

**Comment by jordanmontt** [comment]
@estebanlm I don't know what happened but I made a frivolous commit and now everything is updated. I replaced all `add:withConstraints:` and also deleted the unnecessary accessor methods 🙂

**Comment by estebanlm** [comment]
you left an unused variable at: SpDemoCheckboxesPresenter :)

**Comment by jordanmontt** [comment]
Deleted :)

Do you find something else that can be improved?

**Pull request merged**